frontend/pages/policies/ManagePoliciesPage/components/CalendarEventsModal/CalendarEventsModal.tsx

```diff
@@ -69,30 +69,45 @@ const CalendarEventsModal = ({
   );
   const [showExamplePayload, setShowExamplePayload] = useState(false);
 
-  const validateCalendarEventsFormData = (
-    curFormData: ICalendarEventsFormData
-  ) => {
+  // Used on URL change only when URL error exists and always on attempting to save
+  const validateForm = (newFormData: ICalendarEventsFormData) => {
     const errors: Record<string, string> = {};
-    if (curFormData.enabled) {
-      const { url: curUrl } = curFormData;
-      if (!validURL({ url: curUrl })) {
-        const errorPrefix = curUrl ? `${curUrl} is not` : "Please enter";
-        errors.url = `${errorPrefix} a valid resolution webhook URL`;
-      }
+    const { url: newUrl } = newFormData;
+    if (
+      formData.enabled &&
+      !validURL({ url: newUrl || "", protocol: "http" })
+    ) {
+      const errorPrefix = newUrl ? `${newUrl} is not` : "Please enter";
+      errors.url = `${errorPrefix} a valid resolution webhook URL`;
     }
+
+    setFormErrors(errors);
```

Comment:
The[ pattern of having all state-setting logic in one place](https://github.com/fleetdm/fleet/pull/18396/files#diff-9387a0501f02c597d952bfc15fb967b5d391a104d8fa95d2047a97f51697c9d8L92) was intentional and thought-out. Is there a good reason to move this state setting into this function, while `setFormData` remains in the `onChange` handler as before? If not it would be cleaner to maintain that separation of concerns, that is, this `validate` function should be a pure function that takes in `formData` and returns `errors` with that data

Reply:
I'd agree here. The separation of validation and setting data will help with maintainability in the future.